modules/renter/repair.go

```diff
@@ -154,19 +237,38 @@ func (r *Renter) threadedRepairUploads() {
 			}
 
 			// build host list
+			totalsize := f.pieceSize * uint64(f.erasureCode.NumPieces()) * f.numChunks()
 			var hosts []uploader
-			randHosts := r.hostDB.RandomHosts(f.erasureCode.NumPieces())
-			for i := range randHosts {
+			randHosts := r.hostDB.RandomHosts(f.erasureCode.NumPieces() * 2)
+			for _, h := range randHosts {
+				// probabilistically filter out known bad hosts
+				// unresponsive hosts will be selected with probability 1/(1+nFailures)
+				nFailures, ok := r.blacklist[h.IPAddress]
```

Comment:
don't need to do this now, but blacklist should have both a numerator and denominator. Fail = add 1 to demonminator, success = add 1 to both numerator and denmoinator